prereise/gather/griddata/transmission/geometry.py

```diff
@@ -28,16 +40,20 @@ class Conductor(DataclassWithValidation):
         other parameters if it isn't passed.
     """
 
-    radius: float
+    name: str = None
+    radius: float = None
     material: str = None
     resistance_per_km: float = None
     gmr: float = None
     area: float = None
     permeability: float = None
+    current_limit: float = None
 
     def __post_init__(self):
         # Validate inputs
         self.validate_input_types()  # defined in DataclassWithValidation
+        if self.name is not None:
+            self._get_parameters_from_standard_conductor_table()
```

Comment:
Reading through the logic here, it works fine if a classic conductor is specified by `name` and all the essential corresponding parameters are set, such as `gmr`, from reading the `csv`. However, the code will still go through the logic below just none of them applies. Maybe the ideal behavior is skipping the rest. 

Also, if the user specifies `name` and other parameters simultaneously, other parameters will be overridden by classic ones, which is not revealed by the user. Should we prevent this from happening either by throwing exceptions or warnings?

Thinking a little more on this, as the logic of validating data fields of an object from this class becomes complicated, I'm wondering whether we should build a separate class for classic conductors only. Just a random thought for discussion, current implementation works nicely for now.

Reply:
good point.

Reply:
I refactored it so that if we pass a name, we can't pass any other parameters without creating an exception, and we never check whether anything is missing to infer the missing parameters. I think this should cover us. If the user wants to override a parameter from a named conductor, they can always do so after instantiating the object, as shown in the `test_line_long` test.